src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -37,57 +39,404 @@ const ActorInit En_Cow_InitVars = {
+                this->actor.textId = 0x32C8;
+
+                EnCow_UpdateAnimation(this, globalCtx);
+                return;
+            } else {
+                this->flags |= EN_COW_FLAG_WONT_GIVE_MILK;
+            }
+        } else {
+            this->flags &= ~EN_COW_FLAG_WONT_GIVE_MILK;
+        }
+    }
+
+    if (this->actor.xzDistToPlayer < 150.0f &&
+        ABS_ALT((s16)(this->actor.yawTowardsPlayer - this->actor.shape.rot.y)) < 25000) {
+        if (func_801A5100() == 4) {
+            if (!(gSaveContext.weekEventReg[87] & 1)) {
+                gSaveContext.weekEventReg[87] |= 1;
+                if (func_80114E90() != 0) {
+                    this->actor.textId = 0x32C9;
+                } else {
+                    this->actor.textId = 0x32CA;
+                }
+                this->actor.flags |= 0x10000;
+                func_800B8614(&this->actor, globalCtx, 170.0f);
+                this->actionFunc = EnCow_Talk;
+            }
+        } else {
+            gSaveContext.weekEventReg[87] &= (u8)~1;
+        }
+    }
+
+    EnCow_UpdateAnimation(this, globalCtx);
+}
+
+void EnCow_DoTail(EnCow* this, GlobalContext* globalCtx) {
+    if (this->animationTimer > 0) {
+        this->animationTimer--;
+    } else {
+        this->animationTimer = Rand_ZeroFloat(200.0f) + 40.0f;
+        Animation_Change(&this->skelAnime, &gCowTailIdleAnim, 1.0f, this->skelAnime.curFrame,
+                         Animation_GetLastFrame(&gCowTailIdleAnim), ANIMMODE_ONCE, 1.0f);
+    }
+
+    if (this->actor.xzDistToPlayer < 150.0f &&
+        ABS_ALT((s16)(this->actor.yawTowardsPlayer - this->actor.shape.rot.y)) > 25000) {
+        if (!(this->flags & EN_COW_FLAG_PLAYER_HAS_APPROACHED)) {
+            this->flags |= EN_COW_FLAG_PLAYER_HAS_APPROACHED;
+            if (this->skelAnime.animation == &gCowTailIdleAnim) {
+                this->animationTimer = 0;
+            }
+        }
+    }
+}
 
-extern ColliderCylinderInit D_8099D610;
+void EnCow_Update(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    EnCow* this = THIS;
+    s16 targetX;
+    s16 targetY;
+    Player* player = GET_PLAYER(globalCtx);
 
-extern UNK_TYPE D_060001CC;
-extern UNK_TYPE D_06004010;
-extern UNK_TYPE D_06004348;
+    CollisionCheck_SetOC(globalCtx, &globalCtx2->colChkCtx, &this->colliders[0].base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx2->colChkCtx, &this->colliders[1].base);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Cow/func_8099C290.s")
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Cow/func_8099C328.s")
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Cow/func_8099C41C.s")
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (this->skelAnime.animation == &gCowBodyChewAnim) {
+            Audio_PlayActorSound2(&this->actor, 0x28DF);
```

Comment:
```suggestion
            Audio_PlayActorSound2(&this->actor, NA_SE_EV_COW_CRY);
```
We have sfx ids now thanks to MM3D. You can check them out in `sfx.h`